client/src/main/java/org/eclipse/hono/client/impl/AbstractRequestResponseClient.java

```diff
@@ -699,12 +698,7 @@ protected final void createAndSendRequest(
 
         if (isOpen()) {
             final Message request = createMessage(action, properties);
-            if (payload != null) {
-                if (contentType != null) {
-                    request.setContentType(contentType);
-                }
-                request.setBody(new AmqpValue(payload.getBytes()));
-            }
+            MessageHelper.setPayload(request, contentType, payload);
```

Comment:
did you already verify that a JMS provider based peer will be able to correctly create a JMS TextMessage from this? If we make this change in the code then we also need to change the Credentials, Tenant and Device Registration API as well and document the API change in the release notes ...

Reply:
I didn't create a dedicated test for that yet. However looking at the tests for JMS, I think this is already covered:

https://github.com/eclipse/hono/blob/5527cbb83b5e02bf5fcc8894381e2e33d184f543/tests/src/test/java/org/eclipse/hono/tests/jms/RegistrationTestSupport.java#L168-L178

Reply:
The existing JMS based integration tests only verify if a JMS based *client* can interact with our (non-JMS based) default registry implementation. So, we do not verify if a JMS based implementation of the current Device Registration API would be able to process an AMQP message that contains a Data section instead of an AMQP Value ...

Reply:
But that would use the same JMS functionality and APIs. And in case the JMS implementation would send an AMQP encoded String or byte array instead of a Data section, then Hono would still accept that.

Reply:
Forget my comment on JMS based service implementations. There will not be such a thing because JMS is a client-only technology, i.e. you cannot *implement* an AMQP 1.0 based service endpoint using a JMS provider ...

So I guess you're right in that our existing integration test should be suitable for verifying this. However, we will also need to change the APIs and the device registry so that the response payload is also contained in *Data* sections and an *application/json* content type is set.